pkgs/by-name/bi/bitwarden-desktop/package.nix

```diff
@@ -40,11 +40,22 @@ in buildNpmPackage rec {
 
   patches = [
     ./electron-builder-package-lock.patch
+    ./dont-auto-setup-biometrics.patch
   ];
 
   postPatch = ''
     # remove code under unfree license
     rm -r bitwarden_license
+
+    # The native messaging manifest file (com.8bit.bitwarden.json) and
+    # the generated autostart desktop entry need to have the path to
+    # bitwarden-desktop, but since we use our own electron,
+    # app.getPath("exe") == argv[0] => just electron.
+    # This replaces all instances of app.getPath("exe") with the path
+    # to our wrapper script.
+    pushd apps/desktop/src/
+    find . -type f -name '*.ts' -exec sed -i "s|app.getPath(\"exe\")|\"$out/bin/bitwarden\"|" {} \;
```

Comment:
> This feels brittle.

Understood, that's fair - would you be OK with the manual git patch and substituteAllInPlace approach (as previously described) instead of the current global find/replace?

> I wonder, if you apply the below patch, is this still needed? Reason being that I think this makes the --inherit-argv0 actually have an effect, and I suspect may mean getPath('exe') will be the wrapper instead of the electron.
> Question is then: how is Electron getting the wrong exe path?
> Looking into the source, Electron defers to Chromium which, on Linux, gets the path by resolving /proc/self/exe, so argv0 I guess won’t have an effect on it.

I had a similar thought process initially (aside from the /proc/self/exe part which I never found out) and I believe, even if electron did read from argv0, this solution would still also require a change to electron in nixpkgs, because while electron apps' wrapper scripts *do* set the electron package's argv0 to their wrapper's names, that electron package itself is a wrapper which does *not* set the unwrapped electron binary's argv0 to the one it inherited (see `cat $(nix eval --raw nixpkgs#electron.outPath)/bin/electron`, no exec -a). Unless I'm reading the script wrong; this is what I initially noticed and what prompted me to go for patching bitwarden instead, as any change that would make this work would be far out of the scope of this PR. But feel free to prove me wrong!

Reply:
awesome, that's super clean. I'll add this in rn